mcs/class/corlib/System/Array.cs

```diff
@@ -204,16 +208,31 @@ internal void InternalArray__set_Item<T> (int index, T item)
 				oarray [index] = (object)item;
 				return;
 			}
+			// Do not change this to call SetGenericValue_icall directly, due to special casing in the runtime.
 			SetGenericValueImpl (index, ref item);
 		}
 
 		// CAUTION! No bounds checking!
 		[MethodImplAttribute (MethodImplOptions.InternalCall)]
-		internal extern void GetGenericValueImpl<T> (int pos, out T value);
+		extern static void GetGenericValue_icall<T> (ref Array self, int pos, out T value);
 
 		// CAUTION! No bounds checking!
 		[MethodImplAttribute (MethodImplOptions.InternalCall)]
-		internal extern void SetGenericValueImpl<T> (int pos, ref T value);
+		extern static void SetGenericValue_icall<T> (ref Array self, int pos, ref T value);
+
+		// This is a special case in the runtime.
+		internal void GetGenericValueImpl<T> (int pos, out T value)
```

Comment:
Do we need to protect this from being inlined so that the call gets replaced by an intrinsic?